osu.Game.Rulesets.Taiko/Skinning/Legacy/TaikoLegacySkinTransformer.cs

```diff
@@ -29,119 +32,180 @@ public TaikoLegacySkinTransformer(ISkin skin)
 
         public override Drawable? GetDrawableComponent(ISkinComponentLookup lookup)
         {
-            if (lookup is SkinComponentLookup<HitResult>)
+            switch (lookup)
             {
-                // if a taiko skin is providing explosion sprites, hide the judgements completely
-                if (hasExplosion.Value)
-                    return Drawable.Empty().With(d => d.Expire());
-            }
-
-            if (lookup is TaikoSkinComponentLookup taikoComponent)
-            {
-                switch (taikoComponent.Component)
+                case GlobalSkinnableContainerLookup containerLookup:
                 {
-                    case TaikoSkinComponents.DrumRollBody:
-                        if (GetTexture("taiko-roll-middle") != null)
-                            return new LegacyDrumRoll();
+                    // Modifications for global components.
+                    if (containerLookup.Ruleset == null)
+                        return base.GetDrawableComponent(lookup);
 
+                    // we don't have enough assets to display these components (this is especially the case on a "beatmap" skin).
+                    if (!IsProvidingLegacyResources)
                         return null;
 
-                    case TaikoSkinComponents.InputDrum:
-                        if (hasBarLeft)
-                            return new LegacyInputDrum();
+                    switch (containerLookup.Lookup)
+                    {
+                        case GlobalSkinnableContainers.MainHUDComponents:
+                            return new DefaultSkinComponentsContainer(container =>
+                            {
+                                var combo = container.OfType<LegacyDefaultComboCounter>().FirstOrDefault();
+                                var spectatorList = container.OfType<SpectatorList>().FirstOrDefault();
+                                var leaderboard = container.OfType<DrawableGameplayLeaderboard>().FirstOrDefault();
+
+                                Vector2 pos = new Vector2();
+
+                                if (combo != null)
+                                {
+                                    combo.Anchor = Anchor.BottomLeft;
+                                    combo.Origin = Anchor.BottomLeft;
+                                    combo.Scale = new Vector2(1.28f);
+
+                                    pos += new Vector2(10, -(combo.DrawHeight * 1.56f + 20) * combo.Scale.X);
+                                }
+
+                                if (leaderboard != null)
+                                {
+                                    leaderboard.Anchor = Anchor.BottomLeft;
+                                    leaderboard.Origin = Anchor.BottomLeft;
+                                    leaderboard.Position = pos;
+                                    leaderboard.Height = 170;
```

Comment:
This height can be improved later when the positioning of taiko combo counter is adjusted to match stable (should be on the drum).